plugins/dell-dock/fu-dell-dock-ec-v2.c

```diff
@@ -0,0 +1,1021 @@
+			? "Base"
+			: "Module";
+		if (location_str == NULL)
+			continue;
+
+		/* show the component location */
+		g_debug("#%u: %s located in %s (A: %u I: %u)",
+			i,
+			type_str,
+			location_str,
+			self->dock_info->devices[i].ec_addr_map.arg,
+			self->dock_info->devices[i].ec_addr_map.instance);
+
+		/* show the component version */
+		g_debug("\tVersion32: %08x, Version8: %x %x %x %x",
+			self->dock_info->devices[i].version.version_32,
+			self->dock_info->devices[i].version.version_8[0],
+			self->dock_info->devices[i].version.version_8[1],
+			self->dock_info->devices[i].version.version_8[2],
+			self->dock_info->devices[i].version.version_8[3]);
+
+		/* setup EC component */
+		if (self->dock_info->devices[i].ec_addr_map.device_type ==
+		    EC_v2_DOCK_DEVICE_TYPE_MAIN_EC) {
+			const gchar *ec_version = NULL;
+			ec_version =
+			    g_strdup_printf("%02x.%02x.%02x.%02x",
+					    self->dock_info->devices[i].version.version_8[0],
+					    self->dock_info->devices[i].version.version_8[1],
+					    self->dock_info->devices[i].version.version_8[2],
+					    self->dock_info->devices[i].version.version_8[3]);
+
+			g_debug("\tParsed version %s", ec_version);
+			fu_device_set_version(device, ec_version);
+		}
+
+		/* setup TBT component */
+		if (self->dock_info->devices[i].ec_addr_map.device_type ==
+		    EC_v2_DOCK_DEVICE_TYPE_TBT) {
+			/* Thunderbolt SKU takes a little longer */
+			guint64 tmp = fu_device_get_install_duration(device);
+			fu_device_set_install_duration(device, tmp + 20);
+		}
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_dell_dock_ec_v2_dock_info_cmd(FuDevice *device, GError **error)
+{
+	FuDellDockEcV2 *self = FU_DELL_DOCK_EC_V2(device);
+	gsize length = sizeof(FuDellDockVer2DockInfoStructure);
+
+	g_autoptr(GBytes) data = NULL;
+	g_return_val_if_fail(device != NULL, FALSE);
+
+	/* get dock info over HID */
+	if (!fu_dell_dock_ec_v2_read(device, EC_v2_HID_CMD_GET_DOCK_INFO, length, &data, error)) {
+		g_prefix_error(error, "Failed to query dock info: ");
+		return FALSE;
+	}
+	if (!g_bytes_get_data(data, NULL)) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
+				    "Failed to read dock info");
+		return FALSE;
+	}
+	if (g_bytes_get_size(data) != length) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INVALID_FILE,
+			    "Unexpected dock info size %" G_GSIZE_FORMAT,
+			    g_bytes_get_size(data));
+		return FALSE;
+	}
+	memcpy(self->dock_info, g_bytes_get_data(data, NULL), length);
+	return TRUE;
+}
+
+static gboolean
+fu_dell_dock_ec_v2_dock_data_extract(FuDevice *device, GError **error)
+{
+	FuDellDockEcV2 *self = FU_DELL_DOCK_EC_V2(device);
+	g_autoptr(GString) name = NULL;
+	g_autofree gchar *serial = NULL;
+	const gchar *summary = NULL;
+	g_autofree gchar *board_type_str = NULL;
+
+	/* set FuDevice name */
+	name = g_string_new(self->dock_data->marketing_name);
+	if (name->len > 0)
+		fu_device_set_name(device, name->str);
```

Comment:
has this happened in practice?  I would think you should just be able to do `fu_device_set_name(device, self->dock_data->marketing_name`